src/test/java/com/somesec/crypto/IntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+package com.somesec.crypto;
+
+import com.somesec.crypto.config.ConfigurationResolverImpl;
+import com.somesec.crypto.decrypt.AESDecryption;
+import com.somesec.crypto.decrypt.DecryptionOperation;
+import com.somesec.crypto.encrypt.AESEncryption;
+import com.somesec.crypto.encrypt.EncryptionOperation;
+import com.somesec.crypto.key.DefaultKeyOperationImpl;
+import com.somesec.crypto.key.KeyOperation;
+import org.bouncycastle.crypto.Digest;
+import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.util.encoders.Hex;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.Security;
+
+public class IntegrationTest {
```

Comment:
maybe better to rename the test suite to something related to stream encryption/decryption?